pkg/scheduler/plugins/gang/gang.go

```diff
@@ -44,6 +44,20 @@ func (gp *gangPlugin) Name() string {
 	return "gang"
 }
 
+func backFillEligible(obj interface{}) bool {
+	job := obj.(*api.JobInfo)
+
+	allPending := true
+	for _, task := range job.Tasks {
+		if task.Status != api.Pending {
```

Comment:
if some tasks are `Allocated`, we can not backfill it?

Reply:
is there any callback on select backfill candidate? We discussed one case that only the job with timeout can reuse the idle resources.

Reply:
yes only job with all pending tasks will be backfilling candidate. this possibly can be relaxed later. 

for the second question, it should a lot like what we have for starvation in the other PR following this one.

Reply:
> for the second question, it should a lot like what we have for starvation in the other PR following this one.

This 's not related to reseravation to me; so why handle it in other pr?

Reply:
backfill does not have any timeout configuration. only starvation does.

Reply:
> We discussed one case that only the job with timeout can reuse the idle resources. And the timeout threshold is configurable.
@k82cn  Can you please elaborate the above use case?

Reply:
we need to make sure we have a callback to select which job can be backfilled to improve resource utilization. The job with timeout is a candidate.

Reply:
> yes only job with all pending tasks will be backfilling candidate. this possibly can be relaxed later.

In backfill action, we check eligible firstly before release allocated resources; so if one job has some allocated resource but can not start, this eligible callback will return false, right?